clients/elrond/client.go

```diff
@@ -335,32 +336,47 @@ func (c *client) PerformAction(ctx context.Context, actionID uint64, batch *clie
 	return hash, err
 }
 
-//CheckClientAvailability will check the client availability and will set the metric accordingly
+// CheckClientAvailability will check the client availability and will set the metric accordingly
 func (c *client) CheckClientAvailability(ctx context.Context) error {
-	c.mut.RLock()
-	defer c.mut.RUnlock()
+	c.mut.Lock()
+	defer c.mut.Unlock()
+
 	currentNonce, err := c.GetCurrentNonce(ctx)
 	if err != nil {
-		c.statusHandler.SetStringMetric(bridgeCore.MetricElrondClientStatus, ethElrond.Unavailable.String())
-		c.statusHandler.SetStringMetric(bridgeCore.MetricLastElrondClientError, err.Error())
+		c.setStatusForAvailabilityCheck(ethElrond.Unavailable, err.Error())
+
 		return err
 	}
-	c.statusHandler.SetStringMetric(bridgeCore.MetricLastElrondClientError, "")
 
 	if currentNonce != c.lastNonce {
 		c.retriesAvailabilityCheck = 0
-		c.statusHandler.SetStringMetric(bridgeCore.MetricElrondClientStatus, ethElrond.Available.String())
+		c.lastNonce = currentNonce
 	}
-	c.retriesAvailabilityCheck++
-	if c.retriesAvailabilityCheck >= c.allowDelta {
-		c.statusHandler.SetStringMetric(bridgeCore.MetricElrondClientStatus, ethElrond.Unavailable.String())
-		c.statusHandler.SetStringMetric(bridgeCore.MetricLastElrondClientError,
-			fmt.Sprintf("nonce %d fetched for %d times in a row", currentNonce, c.retriesAvailabilityCheck))
+
+	// if we reached this point we will need to increment the retries counter
+	defer c.incrementRetriesAvailabilityCheck()
+
+	if c.retriesAvailabilityCheck > c.allowDelta {
```

Comment:
```suggestion
	if c.retriesAvailabilityCheck >= c.allowDelta {
```

Reply:
no, not required, the incrementation is done on the defer. Please check unit test :)